modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php

```diff
@@ -0,0 +1,240 @@
+      // images are regenerated.
+      $focal_point_value .= '-' . $random->name();
+      $url = $this->buildUrl($style, $file, $focal_point_value);
+
+      $derivative_images[$style->id()] = [
+        'style' => $style_label,
+        'url' => $url,
+        'image' => [
+          '#theme' => 'image',
+          '#uri' => $url,
+          '#alt' => $this->t('OpenY Focal Point Preview: %label', ['%label' => $style_label]),
+          '#attributes' => [
+            'class' => ['focal-point-derivative-preview-image'],
+          ],
+        ],
+      ];
+
+      $form['openy_focal_point_preview'] = [
+        '#theme' => "openy_focal_point_preview",
+        '#data' => [
+          'derivative_images' => $derivative_images,
+        ],
+      ];
+
+      // Check if this image already has manual crop.
+      $effects = $style->getEffects()->getConfiguration();
+      $manual = array_pop($effects);
+      $crop_type = $manual['data']['crop_type'];
+      $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+      if ($crop) {
+        \Drupal::messenger()->deleteAll();
+
+        \Drupal::messenger()->addWarning('There is manual crop set for this image. It overrides focal point settings');
+        $status_messages = ['#type' => 'status_messages'];
+        $messages_html = drupal_render_root($status_messages);
+        $form['manual_crop_exists'] = [
+          '#markup' => $messages_html,
+        ];
+      }
+
+      // We will display "Focal point updated" message here.
+      $form['messages'] = [
+        '#markup' => '<div id="focal-point-dialog-messages"></div>'
+      ];
+
+      // From FocalPointImageWidget::createFocalPointField().
+      $crop_type = \Drupal::config('focal_point.settings')->get('crop_type');
+      $crop = Crop::findCrop($file->getFileUri(), $crop_type);
+      $image = \Drupal::service('image.factory')->get($file->getFileUri());
+      $width = $image->getWidth();
+      $height = $image->getHeight();
+
+      $anchor = \Drupal::service('focal_point.manager')
+        ->absoluteToRelative($crop->x->value, $crop->y->value, $width, $height);
+      $focal_point_default_value = "{$anchor['x']},{$anchor['y']}";
+
+      $form['focal'] = [
+        '#type' => 'fieldset',
+        '#title' => $this->t('Set focal Point'),
+      ];
+      // Should be unique class for textfield.
+      $focal_point_selector = 'focal-point-' . $style->id();
+
+      $form['focal'][$style->id()]['indicator'] = [
+        '#type' => 'html_tag',
+        '#tag' => 'div',
+        '#attributes' => [
+          'class' => ['focal-point-indicator'],
+          'data-selector' => $focal_point_selector,
+//          'data-delta' => $delta,
+        ],
+      ];
+
+      $form['focal'][$style->id()]['preview'] = [
+        '#theme' => 'image_style',
+        '#width' => $width,
+        '#height' => $height,
+        // @TODO: Avoid hardcoded style name. It comes from field settings.
```

Comment:
when this todo going to be resolved?

Reply:
This is really minor thing. Suggest doing this separately if needed.